pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -245,45 +248,62 @@ func (gce *GCECloud) targetPoolURL(name, region string) string {
 }
 
 func waitForOp(op *compute.Operation, getOperation func() (*compute.Operation, error)) error {
-	pollOp := op
-	consecPollFails := 0
-	for pollOp.Status != "DONE" {
-		var err error
-		time.Sleep(3 * time.Second)
-		pollOp, err = getOperation()
-		if err != nil {
-			if consecPollFails == 2 {
-				// Only bail if we've seen 3 consecutive polling errors.
-				return err
-			}
-			consecPollFails++
-		} else {
-			consecPollFails = 0
-		}
+	if op == nil {
+		return fmt.Errorf("operation must not be nil")
 	}
-	if pollOp.Error != nil && len(pollOp.Error.Errors) > 0 {
+
+	if opIsDone(op) {
+		return getErrorFromCompletedOp(op)
+	}
+
+	var pollOp *compute.Operation
+	var err error
```

Comment:
Nope, just a leftover from when error handling was being done outside. Changed it.